content/posts/2019-08-03---Going-from-ember-to-vue.md

```diff
@@ -0,0 +1,284 @@
+draft: false
+slug: "/posts/going-from-ember-to-vue/"
+category: "Engineering"
+banner: "https://miro.medium.com/max/3840/1*nfvapd86apvGH-hNBYkYuw.png"
+filename: "Going-from-ember-to-vue.md"
+tags:
+  - "Engineering"
+  - "Ember"
+  - "Vue"
+  - "JavaScript"
+  - "Web Development"
+description: "Thinking of looking into another framework other than Ember? React might seem like the obvious choice due to popularity, but consider taking a look at Vue first!"
+---
+
+![Vue.js logo](https://miro.medium.com/max/3840/1*nfvapd86apvGH-hNBYkYuw.png)
+
+If you know me, you know that I love Ember and have worked with it (and the community) for several years.
+But recently, I took another job where I won't be writing it anymore [(I wrote a whole post about this if you're interested)](/posts/a-few-thoughts-on-ember/).
+My first instinct when looking for a new job was to learn React, due to popularity.
+But I found it was hard to jump straight from Ember into React; not impossible, but I felt I had to unlearn some habits that I had formed.
+For the past few months though, I have been working on a Vue project and I have found the leap from Ember to Vue was much more straight forward.
+
+So, I was going to take some time and write about the similarities and differences I have experienced.
+Even if you still love Ember and plan on working with it for a while, I would encourage you to look at Vue just to see what else is out there.
+
+# Some big differences
+
+Much like React, Vue is a component library -- which means you'll be using components for <strong>everything.</strong>
+That means no controllers or route objects.
+When you create a route, you are pointing to a component.
+There is literally nothing special about this component -- it can be literally any component that you want (in fact, you can use it both as a route _and_ component on the page... although, you really shouldn't).
+
+In my opinion, this is a good habit to get into as it will help you transition to something like React down the line if you have to.
+
+## Batteries aren't included
+
+![Batteries](https://i.imgur.com/65SKMnM.jpg?1)
+<aside>Photo by <a href="https://www.pexels.com/@hilaryh">Hilary Halliwell on Pexels</a></aside>
+
+This isn't meant to be a knock on Vue, but it doesn't come with as much stuff out of the box as Ember.
+In fact, this is probably a huge plus.
+[Vue CLI](https://cli.vuejs.org/) allows you to generate a project quickly, but much like Create React App, you are getting a pretty simple UI application with webpack pre-configured for you.
+And yes, webpack over Brocolli is another big difference... but it's probably a good one.
+
+However, anything you want to do in ember, probably already has an equivalent library in Vue.
+And most of these libraries are maintained by core team members, so they feel very official.
+With just a couple more `vue add` commands, you can get routing and state management in your app, so it's really not a _huge_ knock against Vue.
```

Comment:
so is the routing and state management official? (unlike in react?)

what's state management look like? services? redux? other/etc?

Reply:
Can clarify this.

But yes. All of this is found on Vue's official. Hence why the Vuex and Vue Router are under the official vuejs domains:
https://vuex.vuejs.org/
https://router.vuejs.org/

But they are all opt-in.
Vuex is just like Redux. Dispatch an "action", it gets the payload plus a `commit` method, commit calls a `mutation` which gets the state for your whole app, and you update the property you want.